### DISCUSSION TITLE
AGPL license and commercial use

### DISCUSSION THREAD

**Discussion by hholst80** [tobychui/zoraxy#153] [General]
Hi,
I saw in the readme the following statement:

> This project is open-sourced under AGPL. I open-sourced this project so everyone can check for security issues and benefit all users. If you plan to use this project in a commercial environment (which violate the AGPL terms), please contact [toby@imuslab.com](mailto:toby@imuslab.com) for an alternative license.

This is not consistent with how we have seen AGPL software used.  For instance, MongoDB used to have AGPL license and several cloud providers, including AWS, used MongoDB source code to provide their own commercial offering (a managed MongoDB compatible service).

Mongo later changed the license to [Server Side Public License](https://en.wikipedia.org/wiki/Server_Side_Public_License)" (SSPL) to disallow AWS from using it's source code in such cases.

I think to be clear, anyone uses zoraxy to provide their own reverse proxy services (say AWS), it would be difficult argument for a lawyer to make to claim that AGPL did now allow "commercial use"

Neither ChatGPT nor I are lawyers but at least we are in agreement.

![image](https://github.com/tobychui/zoraxy/assets/6200749/dd9a9f52-ca91-4363-8363-1056dd5ac63f)

**Comment by tobychui** [comment]
Hi, what that license terms means is that, if you are planning to use this project in commercial environment which violates the term of AGPL, you need to seek for a commercial license. The phrase "violates the term of AGPL" is referring to the environment not the commercial use. Some of the example cases includes using Zoraxy with proprietary code that is not open source, deploying Zoraxy on your commercial environment with proprietary scripts or provide commercial services to your customers using a close sourced, modified version of Zoraxy. 

I also notice this line bring some confusion to people trying to use this project, so in the [PR for next release](https://github.com/tobychui/zoraxy/pull/152/files), this license text has been changed to the following.

> This project is open-sourced under AGPL. I open-sourced this project so everyone can check for security issues and benefit all users. **This software is intended to be free of charge. If you have acquired this software from a third-party seller, the authors of this repository bears no responsibility for any technical difficulties assistance or support.**

Side notes, GPT model is just an advance version of random text generator, by itself it cannot proof anything. Next time if you want to proof anything important, cite from a reliable source.
